stablehlo/dialect/TypeInference.h

```diff
@@ -216,13 +263,56 @@ LogicalResult inferTriangularSolveOp(
     bool isTransposeAInvalid,
     SmallVectorImpl<ShapedTypeComponents>& inferredReturnShapes);
 
+LogicalResult inferTupleOp(MLIRContext* context, Optional<Location> location,
+                           TypeRange operandTypes,
```

Comment:
How about we change this to `ValueRange val`?

Reply:
Get your point: align with the arg name. Also add the adaptor.